src/credentials/CHIPOpCred.h

```diff
@@ -0,0 +1,184 @@
+                    const uint8_t * deviceCredentialsDER, uint16_t deviceCredentialsLen);
+
+    /**
+     * @brief Find certificate set in the set.
+     *
+     * @param trustedRootId  Authority key identifier of the root certificate-set to be found in the operational credential set.
+     *
+     * @return A pointer to the certificate data On success. Otherwise, NULL if no certificate found.
+     **/
+    ChipCertificateSet * FindCertSet(const CertificateKeyId & trustedRootId) const;
+
+    /**
+     * @return A pointer to the set of certificate-set data entries.
+     **/
+    const ChipCertificateSet * GetOpCredSet() const { return mOpCreds; }
+
+    /**
+     * @return A pointer to the last certificate-set in the set. Returns NULL if certificate set is empty.
+     **/
+    const ChipCertificateSet * GetLastCert() const { return (mOpCredCount > 0) ? &mOpCreds[mOpCredCount - 1] : nullptr; }
+
+    /**
+     * @return Number of certificate-sets loaded into the set.
+     **/
+    uint8_t GetCertCount() const { return mOpCredCount; }
+
+    /**
+     * @brief Retrieve Trusted Root ID / Authority Key ID of a Certificate Set.
+     *
+     * @param certSetIndex  Index of the Certificate-set to be used to search for a Trusted Root ID.
+     *
+     * @return A pointer to the Trusted Root ID on success. Otherwise, nullptr if no Trust Anchor is found.
+     **/
+    const CertificateKeyId * GetTrustedRootId(uint16_t certSetIndex) const;
+
+    /**
+     * @brief Check whether certificate set is in the operational credential set.
+     *
+     * @param cert  Pointer to the ChipCertificateSet structures.
+     *
+     * @return True if certificate is in the set, false otherwise.
+     **/
+    bool IsCertSetInTheOpCredSet(const ChipCertificateSet * cert) const;
+
+    /**
+     * @brief Check whether Trusted Root ID is in the operational credential set.
+     *
+     * @param trustedRoot  Reference to the Trusted Root ID.
+     *
+     * @return True if Trusted Root ID is in the set, false otherwise.
+     **/
+    bool IsTrustedRootIn(const CertificateKeyId & trustedRoot) const;
+
+    /**
+     * @brief Validate CHIP certificate.
+     *
+     * @param trustedRootId    Reference to the Trusted Root ID for the Certificate Set to be used
+     *                          for validation
+     * @param cert              Pointer to the CHIP certificiate to be validated. The certificate is
+     *                          required to be in this set, otherwise this function returns error.
+     * @param context           Certificate validation context.
+     *
+     * @return Returns a CHIP_ERROR on validation or other error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR ValidateCert(const CertificateKeyId & trustedRootId, const ChipCertificateData * cert, ValidationContext & context);
+
+    /**
+     * @brief Find and validate CHIP certificate.
```

Comment:
brief seems to slightly mismatch with the name: I expect `FindValidCert` to "Find a valid certificate".